### PR TITLE
vite

### LINKED ISSUE #544: Move from webpack to vite

### PR DESCRIPTION
**Pull request by LudvigHz**
Resolves #544.

- Rename all react files to .jsx
- Move all frontend assets to be in the same folder
- Remove webpack and add vitejs
- Setup asset loading in django from vite build
- Move from moment to luxon
- fixup! Remove webpack and add vitejs
- Fix eslint config

Production build and dev server seems to work pretty nice.
The only issue as of now, is that HMR/fast refresh doesn't work (it never has). I suspect that this
is due to the error boundary but I'm not sure.

### CODE REVIEW

File: frontend/src/containers/UserApplication/index.jsx [hunk 1]
```diff
@@ -55,11 +53,15 @@ const UserApplication = ({
...
+                <FormatTime format="EEEE dd MMMM, \k\l. HH:mm">
+                  {created_at}
+                </FormatTime>
               </div>
               <div>
                 <SmallDescription> Oppdatert </SmallDescription>
-                <Moment format="dddd Do MMMM, \k\l. HH:mm">{updated_at}</Moment>
+                <FormatTime format="EEEE dd MMMM, \k\l. HH:mm">
+                  {updated_at}
+                </FormatTime>
```
Comment (norbye) [inline frontend/src/containers/UserApplication/index.jsx:64]: It seems the escaping sequence has changed somewhat (screenshot from the group-representative view, that part is fine from the superuser view)
image

File: frontend/src/routes/AdminPageAbakusLeaderView/index.jsx [hunk 2]
```diff
@@ -115,21 +113,21 @@ const AdminPageAbakusLeaderView = () => {
...
-              </Moment>
+              </FormatTime>
             </StatisticsWrapper>
             <StatisticsWrapper>
               <StatisticsName>Redigeringsfrist</StatisticsName>
-              <Moment format="HH:mm:ss dddd Do MMMM">
+              <FormatTime format="HH:mm:ss dddd Do MMMM">
                 {admission.application_deadline}
-              </Moment>
+              </FormatTime>
```
Comment (norbye) [inline frontend/src/routes/AdminPageAbakusLeaderView/index.jsx:130]: A couple of these seem to have changed as well;
image

File: frontend/src/routes/ApplicationForm/FormStructure.jsx [hunk 3]
```diff
@@ -82,9 +80,9 @@ const FormStructure = ({
             Søknaden ble sist registrert
             {myApplication && (
               <StyledSpan bold>
-                <Moment format="dddd Do MMMM, \k\l. HH:mm:ss">
+                <FormatTime format="EEEE dd MMMM, kl. HH:mm:ss">
```
Comment (norbye) [inline frontend/src/routes/ApplicationForm/FormStructure.jsx:83]: Could this be changed from `dd` to `d.`? The `Do` format in moment seems to automatically add the dot, we didn't have padded date strings before and personally I prefer `4.` over `04`

File: frontend/src/routes/ApplicationForm/FormStructure.jsx [hunk 4]
```diff
@@ -95,9 +93,9 @@ const FormStructure = ({
                 <StyledSpan bold> fritt endre søknaden</StyledSpan> din frem til{" "}
                 {admission && (
                   <StyledSpan bold red>
-                    <Moment format="dddd Do MMMM">
+                    <FormatTime format="EEEE dd MMMM">
```
Comment (norbye) [inline frontend/src/routes/ApplicationForm/FormStructure.jsx:96]: `dd`->`d.`?

File: frontend/src/routes/ApplicationForm/FormStructure.jsx [hunk 5]
```diff
@@ -238,14 +236,14 @@ const FormStructure = ({
                   <ApplicationDateInfo>
                     <StyledSpan bold>Søknadsfristen</StyledSpan> er{" "}
                     <StyledSpan bold red>
-                      <Moment format="dddd Do MMMM">
+                      <FormatTime format="EEEE dd MMMM">
```
Comment (norbye) [inline frontend/src/routes/ApplicationForm/FormStructure.jsx:239]: `dd`->`d.`?

File: frontend/src/routes/LandingPage/index.jsx [hunk 6]
```diff
@@ -118,10 +115,14 @@ const ApplicationDateInfo = ({ admission }) => (
     <p>
       Siste frist for å <StyledSpan bold>legge inn en søknad</StyledSpan> er{" "}
       <StyledSpan bold red>
-        <Moment format="dddd Do MMMM">{admission.public_deadline}</Moment>
+        <FormatTime format="EEEE dd. MMMM">
```
Comment (norbye) [inline frontend/src/routes/LandingPage/index.jsx:118]: `dd.`->`d.`? To be consistent with the rest of the date usages, if you agree with the format we had before

File: package.json [hunk 7]
```diff
@@ -44,75 +46,19 @@
...
-    "eslint-config-airbnb-base": "15.0.0",
-    "eslint-config-prettier": "8.5.0",
-    "eslint-config-react-app": "7.0.1",
-    "eslint-import-resolver-webpack": "0.13.2",
-    "eslint-plugin-flowtype": "^8.0.3",
-    "eslint-plugin-import": "2.26.0",
-    "eslint-plugin-jsx-a11y": "6.6.1",
-    "eslint-plugin-prettier": "4.2.1",
-    "eslint-plugin-react": "7.30.1",
-    "eslint-plugin-react-hooks": "^4.6.0",
```
Comment (norbye) [inline package.json:93]: :heart:

File: frontend/src/containers/UserApplication/index.jsx [hunk 8]
```diff
@@ -55,11 +53,15 @@ const UserApplication = ({
             <SmallDescriptionWrapper>
               <div>
                 <SmallDescription> Sendt </SmallDescription>
-                <Moment format="dddd Do MMMM, \k\l. HH:mm">{created_at}</Moment>
+                <FormatTime format="EEEE dd MMMM, \k\l. HH:mm">
```
Comment (norbye) [inline frontend/src/containers/UserApplication/index.jsx:56]: `dd`->`d.`?

File: frontend/src/containers/UserApplication/index.jsx [hunk 9]
```diff
@@ -55,11 +53,15 @@ const UserApplication = ({
...
+                  {created_at}
+                </FormatTime>
               </div>
               <div>
                 <SmallDescription> Oppdatert </SmallDescription>
-                <Moment format="dddd Do MMMM, \k\l. HH:mm">{updated_at}</Moment>
+                <FormatTime format="EEEE dd MMMM, \k\l. HH:mm">
```
Comment (norbye) [inline frontend/src/containers/UserApplication/index.jsx:62]: `dd`->`d.`?

File: frontend/src/containers/UserApplicationAdminView/index.jsx [hunk 10]
```diff
@@ -73,15 +71,15 @@ const UserApplicationAdminView = ({
               <SmallDescriptionWrapper>
                 <div>
                   <SmallDescription> Sendt </SmallDescription>
-                  <Moment format="dddd Do MMMM, \k\l. HH:mm">
+                  <FormatTime format="EEEE dd MMMM, kl. HH:mm">
```
Comment (norbye) [inline frontend/src/containers/UserApplicationAdminView/index.jsx:74]: `dd`->`d.`?

File: frontend/src/containers/UserApplicationAdminView/index.jsx [hunk 11]
```diff
@@ -73,15 +71,15 @@ const UserApplicationAdminView = ({
...
                     {created_at}
-                  </Moment>
+                  </FormatTime>
                 </div>
                 <div>
                   <SmallDescription> Oppdatert </SmallDescription>
-                  <Moment format="dddd Do MMMM, \k\l. HH:mm">
+                  <FormatTime format="EEEE dd MMMM, kl. HH:mm">
```
Comment (norbye) [inline frontend/src/containers/UserApplicationAdminView/index.jsx:80]: `dd`->`d.`?

**Review by norbye — CHANGES_REQUESTED** [review]
Sweet! The luxon-timestrings were changed in some places, but the rest seems to work out great, at least in my dev environment:)

Could you also change https://github.com/webkom/committee-admissions/blob/master/README.md?plain=1#L91 from `yarn watch` to `yarn dev`.

**Comment by norbye** [comment]
> The only issue as of now, is that HMR/fast refresh doesn't work (it never has). I suspect that this
is due to the error boundary but I'm not sure.

Quick question regarding this, does HMR consist of more than the website automatically re-rendering new changes? If no the previous setup did that in my dev environment, but this version doesn't seem to do it (although that might be a bug within vite, as the vite terminal and my dev console both report a hmr update although the site doesn't get updated)

File: frontend/src/containers/UserApplication/index.jsx (hunk 1 above)
Comment (LudvigHz) [reply frontend/src/containers/UserApplication/index.jsx:64]: Yeah, seems I forgot to change it in some places :+1:

File: frontend/src/routes/ApplicationForm/FormStructure.jsx (hunk 3 above)
Comment (LudvigHz) [reply frontend/src/routes/ApplicationForm/FormStructure.jsx:83]: ah, yeah sure

**Comment by LudvigHz** [comment]
> Quick question regarding this, does HMR consist of more than the website automatically re-rendering new changes? If no the previous setup did that in my dev environment, but this version doesn't seem to do it (although that might be a bug within vite, as the vite terminal and my dev console both report a hmr update although the site doesn't get updated)

HMR (react fast refresh in this context) does auto re-rendering yes. I was sure it didn't work properly previously but I might be wrong :sweat_smile:.  The bug in this branch is that even though everything _seems_ to be working, react does not re-render the page. I'm sure this can be fixed through some more debugging. I'll allow you to decide if it's fine to merge this without it and fix it later. (I.e. you need to ctrl+r every time you do a change).

**Comment by norbye** [comment]
I'd rather get rid of webpack than keep HMR so merging this and fixing it later is fine by me

**Review by norbye — APPROVED** [review]
Nice!

**Pull request merged**